server/auth/auth_handler_factory.test.ts

```diff
@@ -23,13 +23,16 @@ import {
 import { SecurityPluginConfigType } from '..';
 import { SecuritySessionCookie } from '../session/security_cookie';
 import { getAuthenticationHandler } from './auth_handler_factory';
+import { AuthType } from '../../common';
+
+const mockBasicAuthType = AuthType.BASIC;
```

Comment:
mock is typically used when creating a `mock` of an object which allows instantiating a complex object, but having any invocations of methods do nothing or do what they are explicitly told to do without invoking the actual implementation. You can update any strings in this test to their enum equivalent inline: `basicauth` -> `AuthType.BASIC`

Reply:
@cwperks When I use AuthType.BASIC, I get this error -> ReferenceError: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables. Invalid variable access: AuthType. This is why I made a new variable.

Reply:
I see, the `mock` prefix to the variable is meaningful [here](https://github.com/jestjs/jest/issues/2567#issuecomment-272015844). 

This change looks good to me. One alternative I can see, is to make a call to `const { AuthType } = require('../../common');` from within `jest.mock('./types', () => { ... }` which also works. It could help avoid making mock variables for every constants within the `AuthType` enum.

Reply:
@cwperks With this I get an eslint error since AuthType is already declared in the upper scope, which I cannot remove since AuthType is referenced later in the test file. Also with eslint, apparently requires is not allowed, so I'm getting an error there too. Can I just move forward with the mock variable as it is?